backend/src/main/java/rocks/inspectit/gepard/agentmanager/connection/service/ConnectionService.java

```diff
@@ -81,20 +81,26 @@ public ConnectionDto getConnection(UUID id) {
    * @return true if the connection matches the query, false otherwise.
    */
   private boolean matchesConnection(Connection connection, QueryConnectionRequest query) {
-    if (query.id() != null && !connection.getId().toString().matches(query.id())) {
-      return false;
-    }
+    boolean matches = true;
 
-    if (query.registrationTime() != null
-        && !connection.getRegistrationTime().toString().matches(query.registrationTime())) {
-      return false;
-    }
+    matches &= regexQueryService.matches(connection.getId().toString(), query.id());
+    matches &=
+        regexQueryService.matches(
+            connection.getRegistrationTime().toString(), query.registrationTime());
 
     if (query.agent() != null) {
-      return matchesAgent(connection.getAgent(), query.agent());
+
+      QueryConnectionRequest.QueryAgentRequest queryAgent = query.agent();
+      Agent agent = connection.getAgent();
+
+      matches &= matchesAgent(agent, queryAgent);
+
+      if (queryAgent.attributes() != null && !queryAgent.attributes().isEmpty()) {
+        matches &= matchesAttributes(agent.getAttributes(), queryAgent.attributes());
+      }
```

Comment:
I believe you can remove this, since you call matchesAttributes() already inside matchesAgent()

Reply:
yop, that´s redudant.